android/sdl_android/src/main/java/com/smartdevicelink/managers/video/VideoStreamManager.java

```diff
@@ -476,145 +476,176 @@ public void run() {
+	private static class SdlMotionEvent {
+		class Pointer {
+			int id;
+			float x;
+			float y;
+			Pointer (int id) {
+				this.id = id;
+				this.x = 0.0f;
+				this.y = 0.0f;
+			}
+			void setCoords(float x, float y) {
+				this.x = x;
+				this.y = y;
+			}
+		}
+
+		private List<Pointer> pointers = new ArrayList<>();
+		private long downTime;
+		private long downTimeOnHMI;
+		private long eventTime;
 
 		SdlMotionEvent(){
-			startOfEvent = SystemClock.uptimeMillis();
+			downTimeOnHMI = 0;
 		}
 
 		/**
-		 * Handles the SDL Touch Event to keep track of pointer status and returns the appropirate
+		 * Handles the SDL Touch Event to keep track of pointer status and returns the appropriate
 		 * Android MotionEvent according to this events status
 		 * @param touchType The SDL TouchType that was received from the module
-		 * @param pointerProperties the parsed pointer properties built from the OnTouchEvent RPC
-		 * @return the correct native Andorid MotionEvent action to dispatch
+		 * @param touchEvent The SDL TouchEvent that was received from the module
+		 * @return the correct native Android MotionEvent action to dispatch
 		 */
-		synchronized int  getMotionEvent(TouchType touchType, MotionEvent.PointerProperties[] pointerProperties){
-			int motionEvent = MotionEvent.ACTION_DOWN;
+		synchronized int getMotionEventAction(TouchType touchType, TouchEvent touchEvent){
+			eventTime = 0;
+			int motionEventAction = -1;
 			switch (touchType){
 				case BEGIN:
-					if(pointerStatuses.size() == 0){
+					if(pointers.size() == 0){
 						//The motion event has just begun
-						motionEvent = MotionEvent.ACTION_DOWN;
-					}else{
-						motionEvent = MotionEvent.ACTION_POINTER_DOWN;
+						motionEventAction = MotionEvent.ACTION_DOWN;
+						downTime = SystemClock.uptimeMillis();
+						downTimeOnHMI = touchEvent.getTimestamps().get(touchEvent.getTimestamps().size() - 1);
+						eventTime = downTime;
+					} else{
+						motionEventAction = MotionEvent.ACTION_POINTER_DOWN | pointers.size() << MotionEvent.ACTION_POINTER_INDEX_SHIFT;
+						eventTime = downTime + touchEvent.getTimestamps().get(touchEvent.getTimestamps().size() - 1) - downTimeOnHMI;
 					}
-					setPointerStatuses(motionEvent, pointerProperties);
+					pointers.add(new Pointer(touchEvent.getId()));
 					break;
 				case MOVE:
-					motionEvent = MotionEvent.ACTION_MOVE;
-					setPointerStatuses(motionEvent, pointerProperties);
-
+					motionEventAction = MotionEvent.ACTION_MOVE;
+					eventTime = downTime + touchEvent.getTimestamps().get(touchEvent.getTimestamps().size() - 1) - downTimeOnHMI;
 					break;
 				case END:
-					//Clears out pointers that have ended
-					setPointerStatuses(MotionEvent.ACTION_UP, pointerProperties);
-
-					if(pointerStatuses.size() == 0){
+					if(pointers.size() > 1){
+						motionEventAction = MotionEvent.ACTION_POINTER_UP | pointers.indexOf(getPointerById(touchEvent.getId())) << MotionEvent.ACTION_POINTER_INDEX_SHIFT;
```

Comment:
For these shifts I think we should check to make sure the pointers are available first. `indexOf` will return a -1 value if it is not contained in the list. This would include if the `getPointerById` returned null.